google-cloud-util/google-cloud-compat-checker/src/main/java/com/google/cloud/compatchecker/GoogleCloudCompatChecker.java

```diff
@@ -43,30 +44,96 @@ protected void logProperty(String s, String s1) {
+    System.out.println("  ALPN is supported: " + openSslAlpnIsSupported);
 
     String osClassifier = (String)osProperties.get(Detector.DETECTED_CLASSIFIER);
 
     boolean compatible = true;
+    boolean warnings = false;
     System.out.println("Checking compatibility...");
     if (supportedClassifiers.contains(osClassifier)) {
-      System.out.println("  This OS + architecture is supported by Forked Tomcat Native.");
+      System.out.println("  [PASS] This OS + architecture is supported.");
     } else {
-      System.out.println("  This OS + architecture is NOT supported by Forked Tomcat Native.");
-      System.out.println("  See http://netty.io/wiki/forked-tomcat-native.html for details.");
+      System.out.println("  [FAIL] This OS + architecture is NOT supported.");
+      compatible = false;
+    }
+    if (bitMode.equals("64")) {
+      System.out.println("  [PASS] 64-bit JVM is supported.");
+    } else {
+      System.out.println("  [FAIL] " + bitMode + "-bit JVM is NOT supported.");
+      compatible = false;
+    }
+    if (openSslIsAvailable) {
+      System.out.println("  [PASS] Open SSL is available");
+    } else {
+      System.out.println("  [FAIL] Open SSL is NOT available");
+      if (OpenSsl.unavailabilityCause() != null) {
+        System.out.println("         Open SSL Unavailability cause:");
+        OpenSsl.unavailabilityCause().printStackTrace(System.out);
+      }
+      compatible = false;
+    }
+    if (openSslAlpnIsSupported) {
+      System.out.println("  [PASS] Open SSL ALPN is supported");
+    } else {
+      System.out.println("  [FAIL] Open SSL ALPN is NOT supported");
       compatible = false;
     }
+    if (javaSpecificationVersion == null) {
+      System.out.println("  [WARN] Couldn't detect java specification version.");
+      warnings = true;
+    } else if (javaSpecificationVersion.equals("1.7")) {
+      System.out.println("  [WARN] gRPC doesn't work on Google App Engine Standard under Java 1.7");
+      warnings = true;
+    }
     if (!compatible) {
       System.out.println("Result: FAIL");
+      System.out.println("  Your environment is not supported by Forked Tomcat Native.");
+      System.out.println("  See http://netty.io/wiki/forked-tomcat-native.html for details.");
+      System.out.println("  This means that you won't be able to use grpc-based APIs, but");
+      System.out.println("  http1-based APIs should still work.");
     } else {
       System.out.println("Result: UNKNOWN (checker implementation not complete)");
+      System.out.println("  Based on what was checked, nothing was identified that would");
+      System.out.println("  prevent you from using grpc-based APIs.");
+      if (warnings) {
+        System.out.println("  However, there were some warnings to watch out for.");
+      }
+    }
+
+    if (!compatible) {
+      return false;
+    } else {
+      return null; // don't want to promise the answer yet
```

Comment:
I believe it is ok to return true. Nobody expects this method to guarantee anything, it checks what it can check and it is ok. There can be more checkers outside, and checks are usually done via &&, so even if this returns true and some other check returns false it will be still false.